util/ssh_client.go

```diff
@@ -29,6 +53,106 @@ func (c *sshClientImpl) ExecCommand(username string, password string, ip string,
 	return string(output), err
 }
 
+func (c *sshClientImpl) UploadFile(username string, password string, ip string, srcFile string, destFile string) error {
+	config := &ssh.ClientConfig{
+		User: username,
+		Auth: []ssh.AuthMethod{
+			ssh.Password(password),
+		},
+	}
+	if !IsIP(ip) {
+		return errors.New("invalid IP address")
+	}
+
+	if IsDir(srcFile) || IsDir(destFile) {
+		return errors.New("Is a directory")
+	}
+	client, err := ssh.Dial("tcp", ip+":22", config)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+
+	conn := (*myssh.Client)(unsafe.Pointer(client))
+	sftp, err := sftp.NewClient(conn)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	defer sftp.Close()
+
+	data, err := ioutil.ReadFile(srcFile)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+
+	f, err := sftp.Create(destFile)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+	if _, err := f.Write([]byte(data)); err != nil {
```

Comment:
Let's split this statement into two. Move the `f.Write` Outside like previous statements.